include/seqan3/core/algorithm/configuration.hpp

```diff
@@ -572,14 +570,11 @@ class configuration : public std::tuple<configs_t...>
 /*!\brief Deduces the correct configuration element type from the passed seqan3::pipeable_config_element.
  * \relates seqan3::configuration
  */
-template <typename derived_t, typename value_t>
-configuration(pipeable_config_element<derived_t, value_t> &&) -> configuration<std::remove_cvref_t<derived_t>>;
-
-/*!\brief Deduces the correct configuration element type from the passed seqan3::pipeable_config_element.
- * \relates seqan3::configuration
- */
-template <typename derived_t, typename value_t>
-configuration(pipeable_config_element<derived_t, value_t> const &) -> configuration<std::remove_cvref_t<derived_t>>;
+template <typename config_t>
+//!\cond
+    requires detail::config_element_specialisation<std::remove_cvref_t<config_t>>
+//!\endcond
+configuration(config_t &&) -> configuration<std::remove_cvref_t<config_t>>;
```

Comment:
This simplification is great.
I think this can pass like this.

One thing I would like to note, (I am not even sure it is legal). Maybe it is possible to remove the `std::remove_cvref_t` in the requires clause of the constructorstor since it is applied by the deduction rule anyways?
so:
```
template <typename config_element_t>
    requires (!std::same_as<config_element_t, configuration>) &&
             detail::config_element_specialisation<config_element_t>
constexpr configuration(config_element_t && config_element) :
    base_type{std::forward<config_element_t>(config_element)}
{}
```

But maybe this is to aggressive type of optimization of code length?